waku/node.go

```diff
@@ -245,3 +242,95 @@ func addPeers(wakuNode *node.WakuNode, addresses []string, protocol protocol.ID)
 		failOnErr(err, "error adding peer")
 	}
 }
+
+func loadPrivateKeyFromFile(path string) (*ecdsa.PrivateKey, error) {
+	src, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	dst := make([]byte, hex.DecodedLen(len(src)))
+	_, err = hex.Decode(dst, src)
+	if err != nil {
+		return nil, err
+	}
+
+	p, err := libp2pcrypto.UnmarshalSecp256k1PrivateKey(dst)
+	if err != nil {
+		return nil, err
+	}
+
+	privKey := (*ecdsa.PrivateKey)(p.(*libp2pcrypto.Secp256k1PrivateKey))
+	privKey.Curve = crypto.S256()
+
+	return privKey, nil
+}
+
+func writePrivateKeyToFile(path string, force bool) error {
+	_, err := os.Stat(path)
+
+	if err == nil && !force {
+		return fmt.Errorf("%s already exists. Use --overwrite to overwrite the file", path)
+	}
+
+	if err := os.Remove(path); err != nil && !os.IsNotExist(err) {
+		return err
+	}
+
+	key, err := crypto.GenerateKey()
+	if err != nil {
+		return err
+	}
+
+	privKey := libp2pcrypto.PrivKey((*libp2pcrypto.Secp256k1PrivateKey)(key))
```

Comment:
I think libp2p crypto supports other private key schemes as well. Shall we rather have an additional option that specifies the PK's scheme and have the client use that instead of only supporting one?

Alternatively, we could bruteforce the creation of `PrivateKey`. When errors on decode we know the scheme must be wrong, then we can try the next.

Reply:
After talking with @jm-clius about this, secp256k1 is considered expected, so no need to check for other PK schemes.